File: src/OS/domain/execution.py
from dataclasses import dataclass, field
from datetime import datetime, timezone
from enum import Enum
from uuid import UUID, uuid4


class ExecutionStatus(Enum):
    """
    Execution状态
    """

    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"


@dataclass
class Execution:
    """
    一次Agent执行过程
    """

    agent_id: UUID
    event_id: UUID
    id: UUID = field(default_factory=uuid4)
    status: ExecutionStatus = ExecutionStatus.PENDING
    created_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))
    started_at: datetime | None = None
    finished_at: datetime | None = None
    error: str | None = None

File: runtime/domain/execution.go
package domain

import "time"

//本轮执行状态，与agent状态独立
type ExecutionStatus string

const (
	ExecutionStatusPending   ExecutionStatus = "pending"
	ExecutionStatusRunning   ExecutionStatus = "running"
	ExecutionStatusCompleted ExecutionStatus = "completed"
	ExecutionStatusFailed    ExecutionStatus = "failed"
)

//agent处理一次投递的记录
type Execution struct {
	ID           ID               `json:"id"`
	AgentID      ID               `json:"agent_id"`
	EventID      ID               `json:"event_id"`
	Status       ExecutionStatus  `json:"status"`
	CreatedAt    time.Time        `json:"created_at"`
	StartedAt    *time.Time       `json:"started_at,omitempty"`
	FinishedAt   *time.Time       `json:"finished_at,omitempty"`
	Error        string           `json:"error,omitempty"`
	AttemptCount uint64           `json:"attempt_count"`
	Result       *ExecutionResult `json:"result,omitempty"`
}

type ExecutionResult struct {
	StateUpdate map[string]any `json:"state_update"`
	Actions     []Action       `json:"actions"`
}

//创建待执行记录
func NewExecution(agentID, eventID ID) Execution {
	return Execution{
		ID:        mustNewID(),
		AgentID:   agentID,
		EventID:   eventID,
		Status:    ExecutionStatusPending,
		CreatedAt: time.Now().UTC(),
	}
}
